mesonbuild/backend/ninjabackend.py

```diff
@@ -2430,7 +2430,9 @@ def guess_external_link_dependencies(self, linker, target, commands, internal):
         # * Meson optimizes libraries from the same build using the symbol extractor.
         #   Just letting ninja use ld generated dependencies would undo this optimization.
         search_dirs = []
+        search_dirs_set = set()
```

Comment:
We have an `OrderedSet()` implementation in mesonlib, perhaps it would be better to use that?

Reply:
My test system has quite some jitter in the test results, but it at least does not seem to be slower using that OrderedSet. And the code is a bit simpler so i changed this to use OrderedSet.